lms/djangoapps/lti_provider/tasks.py

```diff
@@ -0,0 +1,79 @@
+"""
+Asynchronous tasks for the LTI provider app.
+"""
+
+from django.core.exceptions import ObjectDoesNotExist
+import logging
+
+from lms import CELERY_APP
+from lti_provider.models import GradedAssignment
+import lti_provider.outcomes
+from lti_provider.views import parse_course_and_usage_keys
+
+log = logging.getLogger("edx.lti_provider")
+
+
+@CELERY_APP.task
+def send_outcome(points_possible, points_earned, user_id, course_id, usage_id):
+    """
+    Calculate the score for a given user in a problem and send it to the
+    appropriate LTI consumer's outcome service.
+    """
+    if points_earned is None or not points_possible:
+        log.error(
+            "Outcome Service: Invalid Points earned (%s) or points possible (%s)",
+            points_earned, points_possible
+        )
+        return
```

Comment:
That's a fair point. I'll take the check out here, and follow up with a separate PR against the courseware app that adds more checking there.